libnmstate/validator.py

```diff
@@ -170,9 +170,18 @@ def validate_dns(state):
     dns_servers = (
         state.get(DNS.KEY, {}).get(DNS.CONFIG, {}).get(DNS.SERVER, [])
     )
-    if len(dns_servers) > 2:
+    if len(dns_servers) > 3:
+        logging.warning(
+            "The libc resolver may not support more than 3 nameservers."
+        )
+    if (
+        len(dns_servers) > 2
+        and any(n for n in dns_servers if is_ipv6_address(n))
+        and any(n for n in dns_servers if not is_ipv6_address(n))
```

Comment:
I do not understand how `any` fits here.
It may fit if you do `any(is_ipv6_address(n) for n in dns_servers)` and with the same approach to the other.

Reading this with `not all(is_ipv6_address(n) for n in dns_servers) and not all(is_ipv4_address(n) for n in dns_servers)`  makes me understand better what is checked here, but that is just me.

Reply:
`not all ipv6` == `any ipv4`.

Anyway, changed to `not all`.

Reply:
For the `all()` approach. I have to translate from `not all ipv6 and not all ipv4` into `not (all ipv6 or all ipv4)` in my mind to get the whole picture.

For `any()`, it easy to understand, we raise exception when have a ipv4 and ipv6 address defined.

Reply:
I will stick with `any()`